src/helpers/ConfigHelper.ts

```diff
@@ -62,10 +62,25 @@ export class ConfigHelper {
       }
     }
   }
-
+  /**
+   * Checks if current origin matches:
+   *  - config origin exactly
+   *  - variants:
+   *    - config origin without 'www.'
+   *    - config origin with stricter protocol (we should allow http sites to initialize with https but not viceversa)
+   *    - both
+   */
   public static doesCurrentOriginMatchConfigOrigin(configOrigin: string): boolean {
     try {
-      return location.origin === new URL(configOrigin).origin;
+      const configUrl = new URL(configOrigin);
+
+      if (location.origin === configUrl.origin) {
+        return true;
+      }
+
+      const configURLWithout3W = new URL(configUrl.origin.split("www.").join(''));
```

Comment:
Interesting, never though of `.split("www.").join('')` to remove a string. However `.replace("www.", "")` is easier to read.